### ISSUE TITLE
high memory usage and Ultra-long response in 0.12.2

### ISSUE THREAD

**Issue by thetcc** [influxdata/influxdb#6630]
I just used influxdb(0.12.2) collect monitoring data from telegraf using in a production environment. A very small amount of data. But it uses the 10G of physical memory. And the CPU is also a little high. When I stopped influxdb, I found some logs:

[run] 2016/05/15 11:57:46 Signal received, initializing clean shutdown...
[run] 2016/05/15 11:57:46 Waiting for clean shutdown...
[cluster] 2016/05/15 11:57:46 cluster service accept error: network connection closed
[snapshot] 2016/05/15 11:57:46 snapshot listener closed
[shard-precreation] 2016/05/15 11:57:46 Precreation service terminating
[copier] 2016/05/15 11:57:46 copier listener closed
[continuous_querier] 2016/05/15 11:57:46 continuous query service terminating
[retention] 2016/05/15 11:57:46 retention policy enforcement terminating
[monitor] 2016/05/15 11:57:46 shutting down monitor system
[monitor] 2016/05/15 11:57:46 terminating storage of statistics
[http] 2016/05/15 11:57:46 192.168.1.33 - telegraf [14/May/2016:08:05:40 +0800] POST /write?consistency=&db=telegraf&precision=s&rp=rp_telegraf_60days HTTP/1.1 500 49 - InfluxDBClient 97eeb938-1967-11e6-ac05-000000000000 27h52m6.510308596s

[http] 2016/05/15 11:57:46 ::1 - telegraf [14/May/2016:08:00:20 +0800] POST /write?consistency=&db=telegraf&precision=s&rp=rp_telegraf_60days HTTP/1.1 500 49 - InfluxDBClient d9284db3-1966-11e6-ab0a-000000000000 27h57m26.578452913s

[http] 2016/05/15 11:57:46 192.168.1.22 - telegraf [14/May/2016:08:11:00 +0800] POST /write?consistency=&db=telegraf&precision=s&rp=rp_telegraf_60days HTTP/1.1 500 49 - InfluxDBClient 569f4d1c-1968-11e6-ad03-000000000000 27h46m46.61094595s

Why 27h did not respond until I stop the program.

**Comment by e-dard** [comment]
@thetcc thanks for the report. Does this happen on subsequent start/stop cycles? Have you tried upgrading to `0.13`?

This could possibly be a lock issue which was only freed up when you shutdown.

**Comment by thetcc** [comment]
@e-dard  Yes,this happen when  the influxdb is stopped.The logs is continuous.
I have not upgraded to 0.13. I think 0.13 may also be less stable, because just released.
 #6620  seems possible version 0.13 higher memory use.

**Comment by jwilder** [comment]
@thetcc Can you grab a heap dump and profile?

```
curl -o heap.txt "http://localhost:8086/debug/pprof/heap?debug=1"
curl -o goroutine.txt "http://localhost:8086/debug/pprof/goroutine?debug=1"
```

**Comment by thetcc** [comment]
@jwilder  It takes 10G + memory in two days after I run it, because it affects my other programs, I have to stop it. So, now I can not provide a problem dump file. When it just started to use less memory.

**Comment by jwilder** [comment]
@thetcc There was a connection leak issue in an older version of telegraf that caused higher memory usage.   The `27h` write time looks like you might have hit some kind of deadlock.  #6627 fixes one possible deadlock.

Without a profile or more to look into, there isn't much else we can do other than suggest to upgrade.  There are a number of fixes in `0.13` as well as current master that may help.

Going to close this, but we can re-open if you are able to get some profiles when it's consuming a lot of memory.
